Code/Framework/AzCore/AzCore/Math/Vector4.inl

```diff
@@ -28,21 +31,41 @@ namespace AZ
         ;
     }
 
-
     AZ_MATH_INLINE Vector4::Vector4(Simd::Vec4::FloatArgType value)
         : m_value(value)
     {
-        ;
     }
 
     AZ_MATH_INLINE Vector4::Vector4(const Vector2& source)
+    : m_value(Simd::Vec4::FromVec2(source.GetSimdValue()))
+    {
+        m_w = 1.0f;
+    }
+
+    AZ_MATH_INLINE Vector4::Vector4(const Vector2& source, float z) 
+    : m_value(Simd::Vec4::FromVec2(source.GetSimdValue()))
+    {
+        m_z = z;
+        m_w = 1.0f;
+    }
+
+    AZ_MATH_INLINE Vector4::Vector4(const Vector2& source, float z, float w) 
+    : m_value(Simd::Vec4::FromVec2(source.GetSimdValue()))
     {
-        *this = CreateFromVector3(Vector3(source));
+       m_z = z;
+       m_w = w;
     }
```

Comment:
for this operation you are copying the full 128 for sse and then doing two assignment. how does this work in terms of the compiler?